e2e/framework/utils/s3_operations.go

```diff
@@ -22,19 +22,26 @@ type S3Helper struct {
 	s3client *s3.S3
 }
 
+// create a s3 client for specfied endpoint and creds
 func CreateNewS3Helper(endpoint string, keyId string, keySecret string) *S3Helper {
 
 	creds := credentials.NewStaticCredentials(keyId, keySecret, "")
 
+	// create aws s3 config, must use 'other-v2-signature' region for ceph object store
 	awsConfig := aws.NewConfig().
 		WithRegion("other-v2-signature").
 		WithCredentials(creds).
 		WithEndpoint(endpoint).
 		WithS3ForcePathStyle(true).
 		WithDisableSSL(true)
 
+	//create new session
 	ses := session.New()
+
+	//create new s3 client connection
 	c := s3.New(ses, awsConfig)
+
+	// user a custom signer - need to convert newer amz s3 requests to v2 requests since
```

Comment:
It is due to a compatibility issues with the new s3 libraries. , whenever you make a request via aws-sdk-go, it builds a s3 request in this pattern http://{bucketName}.{Endpoint}/.... . When using non amazon object store like ceph,  DNS cannot resolve above url, (it works if your are using actual amazon url  and valid region code).  The sign function is a hack to convert s3 request to older v2 version e.g. http://endpoint/{bucketName}/...
 I found this hack on line and slimed it down to our use.  This should be a one time thing.  Our other option is to ditch was-sdk s3 libraries and find something else. Again using a our own http client is not trivial because we need to figure out all the 'amz' specific headers we need to use.,which are  unfortunately not documented or they are hard to find.